ml_service/util/logger/app_insights_logger.py

```diff
@@ -0,0 +1,188 @@
+import logging
+
+from opencensus.ext.azure import metrics_exporter
+from opencensus.ext.azure.log_exporter import AzureLogHandler
+from opencensus.ext.azure.trace_exporter import AzureExporter
+from opencensus.trace import config_integration
+from opencensus.trace.samplers import ProbabilitySampler
+from opencensus.trace.tracer import Tracer
+
+from opencensus.trace.span import SpanKind
+from opencensus.trace.status import Status
+from opencensus.stats import aggregation as aggregation_module
+from opencensus.stats import measure as measure_module
+from opencensus.stats import stats as stats_module
+from opencensus.stats import view as view_module
+from opencensus.tags import tag_map as tag_map_module
+
+from ml_service.util.env_variables import Env
+from ml_service.util.logger.logger_interface import (
+    LoggerInterface,
+    ObservabilityAbstract,
+    Severity,
+)
+
+
+class AppInsightsLogger(LoggerInterface, ObservabilityAbstract):
+    def __init__(self, run):
+        print('Initializing the AppInsightsLogger')
+        self.env = Env()
+        self.run_id = self.get_run_id_and_set_context(run)
+
+        # Prepare integrations and initialize tracer
+        config_integration.trace_integrations(['httplib', 'logging'])
+        texporter = AzureExporter(connection_string=self.
+                                  env.app_insights_connection_string)
+        texporter.add_telemetry_processor(self.callback_function)
+        self.tracer = Tracer(
+            exporter=texporter,
+            sampler=ProbabilitySampler(self.env.trace_sampling_rate)
+        )
+
+        # Create AppInsights Handler and set log format
+        self.logger = logging.getLogger(__name__)
+        self.logger.setLevel(
+            getattr(logging, self.env.log_level.upper(), "WARNING"))
```

Comment:
Just curious why default is set to WARNING when we set DEBUG/INFO in env? Do we want to set the lowest level instead?